### ISSUE TITLE
SWT layout in Choose Workspace Dialog broken in I20240109-1800

### ISSUE THREAD

**Issue by ptziegler** [eclipse-platform/eclipse.platform.ui#1492]
Let's make sure issue is not already fixed in latest builds first.

- [x] I verified I can reproduce this issue against [latest Integration Build of Eclipse SDK](https://download.eclipse.org/eclipse/downloads/)

## Steps to reproduce

Download and launch the latest SDK nightly build.

Before (4.30):

![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/70652978/8eaf9ebe-8e8d-4ef7-aa7c-87b295418758)

After (4.31I):

![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/70652978/5b60b940-8897-4863-a5f4-559e0488f58f)

I believe this is a result of https://github.com/eclipse-platform/eclipse.platform.ui/pull/1477, which shows the absolute path if only a relative path is given. The problem arises when the path is already fully qualified, which makes the empty field look out-of-place.

## Tested under this environment:

* Eclipse SDK
* Version: 2024-03 (4.31)
* Build id: I20240109-1800
* OS: Windows 10, v.10.0, x86_64 / win32
* Java vendor: Eclipse Adoptium
* Java runtime version: 21.0.1+12-LTS
* Java version: 21.0.1

## Community

- [x] I understand reporting an issue to this OSS project does not mandate anyone to fix it. Other contributors may consider the issue, or not, at their own convenience. The most efficient way to get it fixed is that I fix it myself and contribute it back as a good quality patch to the project.

**Comment by ptziegler** [comment]
Whoops, I just noticed that it's already handled by #1491

**Comment by ptziegler** [comment]
I'll close this comment again, given that the issue is already know and tracked via the other issues, to avoid confusion.

**Comment by merks** [comment]
Thanks for reporting though!  👍

**Comment by laeubi** [comment]
By the way I find such dedicated reports much more useful than reopen the original issue, ist usually better to just link the one...

**Comment by iloveeclipse** [comment]
https://github.com/eclipse-platform/eclipse.platform.ui/pull/1502 was merged.
All: please check in tomorrow build how it looks like and which remaining issues we have to fix (if any).

**Comment by ptziegler** [comment]
Looking good with the I20240115-1800 👌
Even things like whitespaces in the filepath seem to work without problem again.

![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/70652978/7ed0c5b2-c5d3-4668-9756-fe686711e78b)

**Comment by HeikoKlare** [comment]
On Windows, I now have the problem that when entering anything unresolvable before a double colon, the input field freezes. For example, typing "DD:" freezes the input field. The more likely case is that you, by accident, type something at the beginning of the input field when it already contains an absolute path. Then the input field freezes as well.
When you then click "Launch", startup fails because of the unresolvable path and a faulty entry is stored in the IDE preferences (configuration/.settings/org.eclipse.ui.ide.prefs), e.g.:
```
RECENT_WORKSPACES=DD\:\\dev\\eclipse\\platform-sdk\\ws\nD\:\\dev\\eclipse\\platform-sdk\\ws
```
This results in the IDE not starting anymore and requiring you to fix the preferences file manually:
![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/755472/151c8127-f02f-48f6-b758-28ff8a127f93)

**Comment by HeikoKlare** [comment]
Just verified with today's I-Build that my oberservations mentioned in the preceding comment have been resolved by #1522.
